File: msgraph-mail/ruby/graphrubyv4/models/message_rule.rb
require 'microsoft_kiota_abstractions'
require_relative '../graphrubyv4'
require_relative './models'

module Graphrubyv4
    module Models
        class MessageRule < Graphrubyv4::Models::Entity
            include MicrosoftKiotaAbstractions::Parsable
            ## 
            # The actions property
            @actions
            ## 
            # The conditions property
            @conditions
            ## 
            # The display name of the rule.
            @display_name
            ## 
            # The exceptions property
            @exceptions
            ## 
            # Indicates whether the rule is in an error condition. Read-only.
            @has_error
            ## 
            # Indicates whether the rule is enabled to be applied to messages.
            @is_enabled
            ## 
            # Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
            @is_read_only
            ## 
            # Indicates the order in which the rule is executed, among other rules.
            @sequence
            ## 
            ## Gets the actions property value. The actions property
            ## @return a message_rule_actions
            ## 
            def actions
                return @actions
            end
            ## 
            ## Sets the actions property value. The actions property
            ## @param value Value to set for the actions property.
            ## @return a void
            ## 
            def actions=(value)
                @actions = value
            end
            ## 
            ## Gets the conditions property value. The conditions property
            ## @return a message_rule_predicates
            ## 
            def conditions
                return @conditions
            end
            ## 
            ## Sets the conditions property value. The conditions property
            ## @param value Value to set for the conditions property.
            ## @return a void
            ## 
            def conditions=(value)
                @conditions = value
            end
            ## 
            ## Instantiates a new MessageRule and sets the default values.
            ## @return a void
            ## 
            def initialize()
                super
            end
            ## 
            ## Creates a new instance of the appropriate class based on discriminator value
            ## @param parse_node The parse node to use to read the discriminator value and create the object
            ## @return a message_rule
            ## 
            def self.create_from_discriminator_value(parse_node)
                raise StandardError, 'parse_node cannot be null' if parse_node.nil?
                return MessageRule.new
            end
            ## 
            ## Gets the displayName property value. The display name of the rule.
            ## @return a string
            ## 
            def display_name
                return @display_name
            end
            ## 
            ## Sets the displayName property value. The display name of the rule.
            ## @param value Value to set for the displayName property.
            ## @return a void
            ## 
            def display_name=(value)
                @display_name = value
            end
            ## 
            ## Gets the exceptions property value. The exceptions property
            ## @return a message_rule_predicates
            ## 
            def exceptions
                return @exceptions
            end
            ## 
            ## Sets the exceptions property value. The exceptions property
            ## @param value Value to set for the exceptions property.
            ## @return a void
            ## 
            def exceptions=(value)
                @exceptions = value
            end
            ## 
            ## The deserialization information for the current model
            ## @return a i_dictionary
            ## 
            def get_field_deserializers()
                return super.merge({
                    "actions" => lambda {|n| @actions = n.get_object_value(lambda {|pn| Graphrubyv4::Models::MessageRuleActions.create_from_discriminator_value(pn) }) },
                    "conditions" => lambda {|n| @conditions = n.get_object_value(lambda {|pn| Graphrubyv4::Models::MessageRulePredicates.create_from_discriminator_value(pn) }) },
                    "displayName" => lambda {|n| @display_name = n.get_string_value() },
                    "exceptions" => lambda {|n| @exceptions = n.get_object_value(lambda {|pn| Graphrubyv4::Models::MessageRulePredicates.create_from_discriminator_value(pn) }) },
                    "hasError" => lambda {|n| @has_error = n.get_boolean_value() },
                    "isEnabled" => lambda {|n| @is_enabled = n.get_boolean_value() },
                    "isReadOnly" => lambda {|n| @is_read_only = n.get_boolean_value() },
                    "sequence" => lambda {|n| @sequence = n.get_number_value() },
                })
            end
            ## 
            ## Gets the hasError property value. Indicates whether the rule is in an error condition. Read-only.
            ## @return a boolean
            ## 
            def has_error
                return @has_error
            end
            ## 
            ## Sets the hasError property value. Indicates whether the rule is in an error condition. Read-only.
            ## @param value Value to set for the hasError property.
            ## @return a void
            ## 
            def has_error=(value)
                @has_error = value
            end
            ## 
            ## Gets the isEnabled property value. Indicates whether the rule is enabled to be applied to messages.
            ## @return a boolean
            ## 
            def is_enabled
                return @is_enabled
            end
            ## 
            ## Sets the isEnabled property value. Indicates whether the rule is enabled to be applied to messages.
            ## @param value Value to set for the isEnabled property.
            ## @return a void
            ## 
            def is_enabled=(value)
                @is_enabled = value
            end
            ## 
            ## Gets the isReadOnly property value. Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
            ## @return a boolean
            ## 
            def is_read_only
                return @is_read_only
            end
            ## 
            ## Sets the isReadOnly property value. Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
            ## @param value Value to set for the isReadOnly property.
            ## @return a void
            ## 
            def is_read_only=(value)
                @is_read_only = value
            end
            ## 
            ## Gets the sequence property value. Indicates the order in which the rule is executed, among other rules.
            ## @return a integer
            ## 
            def sequence
                return @sequence
            end
            ## 
            ## Sets the sequence property value. Indicates the order in which the rule is executed, among other rules.
            ## @param value Value to set for the sequence property.
            ## @return a void
            ## 
            def sequence=(value)
                @sequence = value
            end
            ## 
            ## Serializes information the current object
            ## @param writer Serialization writer to use to serialize this model
            ## @return a void
            ## 
            def serialize(writer)
                raise StandardError, 'writer cannot be null' if writer.nil?
                super
                writer.write_object_value("actions", @actions)
                writer.write_object_value("conditions", @conditions)
                writer.write_string_value("displayName", @display_name)
                writer.write_object_value("exceptions", @exceptions)
                writer.write_boolean_value("hasError", @has_error)
                writer.write_boolean_value("isEnabled", @is_enabled)
                writer.write_boolean_value("isReadOnly", @is_read_only)
                writer.write_number_value("sequence", @sequence)
            end
        end
    end
end

File: msgraph-mail/php/src/Models/MessageRule.php
<?php

namespace Microsoft\Graph\Models;

use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

class MessageRule extends Entity implements Parsable 
{
    /**
     * @var MessageRuleActions|null $actions The actions property
    */
    private ?MessageRuleActions $actions = null;
    
    /**
     * @var MessageRulePredicates|null $conditions The conditions property
    */
    private ?MessageRulePredicates $conditions = null;
    
    /**
     * @var string|null $displayName The display name of the rule.
    */
    private ?string $displayName = null;
    
    /**
     * @var MessageRulePredicates|null $exceptions The exceptions property
    */
    private ?MessageRulePredicates $exceptions = null;
    
    /**
     * @var bool|null $hasError Indicates whether the rule is in an error condition. Read-only.
    */
    private ?bool $hasError = null;
    
    /**
     * @var bool|null $isEnabled Indicates whether the rule is enabled to be applied to messages.
    */
    private ?bool $isEnabled = null;
    
    /**
     * @var bool|null $isReadOnly Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
    */
    private ?bool $isReadOnly = null;
    
    /**
     * @var int|null $sequence Indicates the order in which the rule is executed, among other rules.
    */
    private ?int $sequence = null;
    
    /**
     * Instantiates a new MessageRule and sets the default values.
    */
    public function __construct() {
        parent::__construct();
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return MessageRule
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): MessageRule {
        return new MessageRule();
    }

    /**
     * Gets the actions property value. The actions property
     * @return MessageRuleActions|null
    */
    public function getActions(): ?MessageRuleActions {
        return $this->actions;
    }

    /**
     * Gets the conditions property value. The conditions property
     * @return MessageRulePredicates|null
    */
    public function getConditions(): ?MessageRulePredicates {
        return $this->conditions;
    }

    /**
     * Gets the displayName property value. The display name of the rule.
     * @return string|null
    */
    public function getDisplayName(): ?string {
        return $this->displayName;
    }

    /**
     * Gets the exceptions property value. The exceptions property
     * @return MessageRulePredicates|null
    */
    public function getExceptions(): ?MessageRulePredicates {
        return $this->exceptions;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return array_merge(parent::getFieldDeserializers(), [
            'actions' => fn(ParseNode $n) => $o->setActions($n->getObjectValue([MessageRuleActions::class, 'createFromDiscriminatorValue'])),
            'conditions' => fn(ParseNode $n) => $o->setConditions($n->getObjectValue([MessageRulePredicates::class, 'createFromDiscriminatorValue'])),
            'displayName' => fn(ParseNode $n) => $o->setDisplayName($n->getStringValue()),
            'exceptions' => fn(ParseNode $n) => $o->setExceptions($n->getObjectValue([MessageRulePredicates::class, 'createFromDiscriminatorValue'])),
            'hasError' => fn(ParseNode $n) => $o->setHasError($n->getBooleanValue()),
            'isEnabled' => fn(ParseNode $n) => $o->setIsEnabled($n->getBooleanValue()),
            'isReadOnly' => fn(ParseNode $n) => $o->setIsReadOnly($n->getBooleanValue()),
            'sequence' => fn(ParseNode $n) => $o->setSequence($n->getIntegerValue()),
        ]);
    }

    /**
     * Gets the hasError property value. Indicates whether the rule is in an error condition. Read-only.
     * @return bool|null
    */
    public function getHasError(): ?bool {
        return $this->hasError;
    }

    /**
     * Gets the isEnabled property value. Indicates whether the rule is enabled to be applied to messages.
     * @return bool|null
    */
    public function getIsEnabled(): ?bool {
        return $this->isEnabled;
    }

    /**
     * Gets the isReadOnly property value. Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
     * @return bool|null
    */
    public function getIsReadOnly(): ?bool {
        return $this->isReadOnly;
    }

    /**
     * Gets the sequence property value. Indicates the order in which the rule is executed, among other rules.
     * @return int|null
    */
    public function getSequence(): ?int {
        return $this->sequence;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        parent::serialize($writer);
        $writer->writeObjectValue('actions', $this->getActions());
        $writer->writeObjectValue('conditions', $this->getConditions());
        $writer->writeStringValue('displayName', $this->getDisplayName());
        $writer->writeObjectValue('exceptions', $this->getExceptions());
        $writer->writeBooleanValue('hasError', $this->getHasError());
        $writer->writeBooleanValue('isEnabled', $this->getIsEnabled());
        $writer->writeBooleanValue('isReadOnly', $this->getIsReadOnly());
        $writer->writeIntegerValue('sequence', $this->getSequence());
    }

    /**
     * Sets the actions property value. The actions property
     * @param MessageRuleActions|null $value Value to set for the actions property.
    */
    public function setActions(?MessageRuleActions $value): void {
        $this->actions = $value;
    }

    /**
     * Sets the conditions property value. The conditions property
     * @param MessageRulePredicates|null $value Value to set for the conditions property.
    */
    public function setConditions(?MessageRulePredicates $value): void {
        $this->conditions = $value;
    }

    /**
     * Sets the displayName property value. The display name of the rule.
     * @param string|null $value Value to set for the displayName property.
    */
    public function setDisplayName(?string $value): void {
        $this->displayName = $value;
    }

    /**
     * Sets the exceptions property value. The exceptions property
     * @param MessageRulePredicates|null $value Value to set for the exceptions property.
    */
    public function setExceptions(?MessageRulePredicates $value): void {
        $this->exceptions = $value;
    }

    /**
     * Sets the hasError property value. Indicates whether the rule is in an error condition. Read-only.
     * @param bool|null $value Value to set for the hasError property.
    */
    public function setHasError(?bool $value): void {
        $this->hasError = $value;
    }

    /**
     * Sets the isEnabled property value. Indicates whether the rule is enabled to be applied to messages.
     * @param bool|null $value Value to set for the isEnabled property.
    */
    public function setIsEnabled(?bool $value): void {
        $this->isEnabled = $value;
    }

    /**
     * Sets the isReadOnly property value. Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
     * @param bool|null $value Value to set for the isReadOnly property.
    */
    public function setIsReadOnly(?bool $value): void {
        $this->isReadOnly = $value;
    }

    /**
     * Sets the sequence property value. Indicates the order in which the rule is executed, among other rules.
     * @param int|null $value Value to set for the sequence property.
    */
    public function setSequence(?int $value): void {
        $this->sequence = $value;
    }

}
